src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1598,19 +1608,34 @@ impl Decimal {
         if inner_idx != 0 {
             d.word_buf[word_idx] = word * TEN_POW[DIGITS_PER_WORD as usize - inner_idx];
         }
-
-        if end_idx + 1 < bs.len() && (bs[end_idx] == b'e' || bs[end_idx] == b'E') {
+        if end_idx < bs.len() && (bs[end_idx] == b'e' || bs[end_idx] == b'E') {
```

Comment:
why not +1 any more?

Reply:
The end_idx could be `bs.len-1` here, and after we fixed `bytes_to_int_without_context`, this case should return a truncated value instead of ok.